### ISSUE TITLE
Singularity image for vanilla Biocontainers

### ISSUE THREAD

**Issue by drpatelh** [BioContainers/containers#391]
Hi! Hope you are all doing well 🙂 

With the recent stable release of the more modular Nextflow DSL2 we are now migrating to using Biocontainers per process which is working amazingly well. As a proof-of-principle I am currently re-writing the [nf-core/rnaseq](https://github.com/nf-core/rnaseq/tree/dev) pipeline in DSL2. The fact that you have exactly the same images available in Docker and Singularity is also great because we can just pull the relevant container depending on the users local set-up, an example of this logic is [here](https://github.com/nf-core/modules/blob/ff0fe1a910cbaf7e45bb9df4997ecb154eed95d2/software/fastqc/main.nf#L14-L19). This works really well for tool-specific containers but I also wanted to be able to use a vanilla Linux Biocontainer that is available as both a Docker and Singularity container. This would be used in instances where a particular module only requires Linux utilities e.g. `awk`, `paste`, `cat` etc. I done a bit of digging around and decided to use [the `biocontainers/biocontainers:v1.2.0_cv1` container available on Dockerhub](https://github.com/nf-core/rnaseq/blob/0a8040cb6a48572af4a17726c702efb5d0b73bdd/modules/local/process/gunzip.nf#L15), however, I couldn't find an associated Singularity image @ `https://depot.galaxyproject.org/singularity/`.

In short, it would be great if you could point me to any Biocontainers that are as vanilla Linux as possible, and available as both Docker and Singularity images. I did look at for example the `sed` containers but they contain a very basic Linux install (Busybox) with older versions of standard tools like `gunzip` and `tar`.

Thanks in advance!

**Comment by osallou** [comment]
Hi, we can for sure make base image available as singulirity image, but why not using an ubuntu base image?

Biocontainers base adds some python, gcc etc stuff that will increase image size and not useful for basic linux tasks.

**Comment by drpatelh** [comment]
Hi @osallou! Thanks for the rapid response! I could indeed use an Ubuntu base image but I thought I would try and exclusively use Biocontainer images. Also, I would still need an Ubuntu base image in Singularity too. If that is available then I guess there is no reason to use the Biocontainer as you suggested.

**Comment by osallou** [comment]
there is no real added value to use our base image as a pure image, it is more to build most of containers on same base (though they do not all use it) with stuff preinstalled for common builds etc...
ubuntu is for sure available as singularity in singularity official registry

**Comment by ewels** [comment]
Hi @osallou - we unfortunately can't use the singularity official registry as it requires an API key to be fetched manually every 30 (?) days, which doesn't play well with our automation across 10s of repos and 100s of users. The simpler Singularity downloads from the galaxyproject servers works brilliantly though 🚀

**Comment by osallou** [comment]
oh, they do really require an api key do to so? no so cool....

**Comment by osallou** [comment]
our base image is available here

https://containers.biocontainers.pro/s3/SingImgsRepo/biocontainers/v1.2.0_cv1/

**Comment by drpatelh** [comment]
Thank you very much @osallou ! 😎 This will do perfectly fine. I am assuming this will be a persistent URL just so that we can ensure reproducibility?

**Comment by osallou** [comment]
Humm persistent... yes while ebi cloud is available, can host us and there is no s3 data loss... ;-)

We though few years ago dockerhub would be persistent, well it is not anymore... and we are considering moving to quay.io till ....  ;-)

But at least for mid term we can consider it as a stable resource.

**Comment by drpatelh** [comment]
Agreed! Who knows what is going to happen tomorrow with the way things have gone this year! Mid-term is a good enough answer for me given the circumstances :)

**Comment by drpatelh** [comment]
@osallou I am trying to fetch the Biocontainer below but getting a `Service Unavailable` error:

`https://containers.biocontainers.pro/s3/SingImgsRepo/biocontainers/v1.2.0_cv1/biocontainers_v1.2.0_cv1.img`

Have these been moved or is the hosting service down? Thanks!

**Comment by osallou** [comment]
humm just an access issue, thanks for info I gonna fix this

**Comment by osallou** [comment]
@drpatelh this is fixed, sorry for issue

**Comment by drpatelh** [comment]
You just made my day @osallou ! Thanks for the quick fix! :bowtie:

**Comment by drpatelh** [comment]
Hi @osallou! Happy new year! Hope you and loved ones are keeping well.

Sorry, to bother you again but we are seeing network issues again whilst trying to pull the Biocontainer I mentioned in https://github.com/BioContainers/containers/issues/391#issuecomment-743244873. Reported in https://github.com/nf-core/rnaseq/issues/549

**Comment by osallou** [comment]
Arg seems s3 gateway stopped again...
Will fix that tomorrow morning.

Don t understand what is the issue, i need a way to fix this

**Comment by osallou** [comment]
Our storage provider does not support s3 indexing (acting as a base file server) sp we need to put in the middle a gateway to do the job.

**Comment by osallou** [comment]
@drpatelh , server reachable

I did an update to force restarts in case of failure, expecting this will be fine

**Comment by drpatelh** [comment]
Thank you very much for the fix @osallou ! Much appreciated.

**Comment by drpatelh** [comment]
Hi @osallou ! Hope you are well and Happy New Year! 

Pasting the link below in a browser is returning:

https://containers.biocontainers.pro/s3/SingImgsRepo/biocontainers/v1.2.0_cv1/biocontainers_v1.2.0_cv1.img

```
InvalidAccessKeyId: 
	status code: 403, request id: tx000000000000004304341-0061e7daa7-126e86c6-default, host id: 
```

Would it be possible to give things a kick please?

**Comment by osallou** [comment]
Hi,
we faced issues with our cloud provider, I had to reconfigure some stuff. It is ok now

**Comment by drpatelh** [comment]
Absolutely amazing!!

**Comment by jan-hartmann-uksh** [comment]
I think its down again. Maybe kick it again? :)

**Comment by osallou** [comment]
up again thanks

there were some upgrades on their cloud, I suppose they broke stuff again.....
